src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -197,7 +197,7 @@ RazorProjectEngine CreateProjectEngine()
 
                 return _projectEngineFactoryProvider.Create(configuration, rootDirectoryPath, builder =>
                 {
-                    builder.SetRootNamespace(HostProject.RootNamespace);
+                    builder.SetRootNamespace(HostProject.Configuration.RootNamespace);
```

Comment:
`HostProject.Configuration` is already captured in a local above that can be used here (like the line below):

```suggestion
                    builder.SetRootNamespace(configuration.RootNamespace);
```